Source/WebCore/style/values/primitives/StylePosition.cpp

```diff
@@ -85,6 +162,76 @@ static auto resolveKeyword(CSS::Keyword::Center, const BuilderState&) -> LengthP
     return 50_css_percentage;
 }
 
+// MARK: Mapped value resolution
```

Comment:
There's a lot of indirection here, and I wonder if it's necessary?

Reply:
I found it to be the most clear way of implementing it without duplicating a bunch of logic. From a performance perspective, it all inlines and compiles down to an optimal single set of branches. (Also, when the block/inline keywords are involved, they use the same logic.

Reply:
Actually, with the use of simpler writing mode check, I can reduce the cognitive overhead here a bit.